app/models/product_drive_participant.rb

```diff
@@ -31,6 +31,10 @@ def volume
     donations.map { |d| d.line_items.total }.reduce(:+)
   end
 
+  def volume_by_product_drive(product_drive_id)
+    donations.by_product_drive(product_drive_id).map { |d| d.line_items.total }.reduce(:+)
```

Comment:
Minor: I think in general we'd want to default to `0` if we _somehow_ ended up with an empty array. Is that right? If so, would make sense to use `sum` rather than `reduce` to get that default value simply.

```suggestion
    donations.by_product_drive(product_drive_id).map { |d| d.line_items.total }.sum
```